test/Npgsql.Tests/SystemTransactionTests.cs

```diff
@@ -130,21 +162,106 @@ public void TwoConnections()
         }
 
         [Test]
+        public void TwoConsecutiveConnections()
+        {
+            using (var scope = new TransactionScope())
+            {
+                using (var conn1 = OpenConnection(ConnectionStringEnlistOn))
+                {
+                    Assert.That(conn1.ExecuteNonQuery(@"INSERT INTO data (name) VALUES ('test1')"), Is.EqualTo(1),
+                        "Unexpected first insert rowcount");
+                }
+
+                using (var conn2 = OpenConnection(ConnectionStringEnlistOn))
+                {
+                    Assert.That(conn2.ExecuteNonQuery(@"INSERT INTO data (name) VALUES ('test2')"), Is.EqualTo(1),
+                        "Unexpected second insert rowcount");
+                }
+
+                // Consecutive connections used in same scope should not promote the
+                // transaction to distributed.
+                AssertNoPreparedTransactions();
+                scope.Complete();
+            }
+            AssertNumberOfRows(2);
+        }
+
+        [Test]
         public void TwoConnectionsRollback()
         {
             using (new TransactionScope())
             using (var conn1 = OpenConnection(ConnectionStringEnlistOn))
             using (var conn2 = OpenConnection(ConnectionStringEnlistOn))
             {
-                Assert.That(conn1.ExecuteNonQuery(@"INSERT INTO data (name) VALUES ('test1')"), Is.EqualTo(1));
-                Assert.That(conn2.ExecuteNonQuery(@"INSERT INTO data (name) VALUES ('test2')"), Is.EqualTo(1));
+                Assert.That(conn1.ExecuteNonQuery(@"INSERT INTO data (name) VALUES ('test1')"), Is.EqualTo(1),
+                    "Unexpected first insert rowcount");
+                Assert.That(conn2.ExecuteNonQuery(@"INSERT INTO data (name) VALUES ('test2')"), Is.EqualTo(1),
+                    "Unexpected second insert rowcount");
             }
             // TODO: There may be a race condition here, where the prepared transaction above still hasn't committed.
             AssertNoPreparedTransactions();
             AssertNumberOfRows(0);
         }
 
         [Test]
+        public void DistributedRollback()
+        {
+            var disposedCalled = false;
+            var tx = new TransactionScope();
+            try
+            {
+                using (var conn1 = OpenConnection(ConnectionStringEnlistOn))
+                {
+                    Assert.That(conn1.ExecuteNonQuery(@"INSERT INTO data (name) VALUES ('test1')"), Is.EqualTo(1),
+                        "Unexpected first insert rowcount");
+
+                    ForceEscalationToDistributedTx.Escalate(true);
```

Comment:
OK. To me this test seems very similar to TwoConnectionsRollback - except in that the enlistment triggers the rollback (via `ForceRollback()`) and not the user (by not calling `TransactionScope.Complete()`). I'm not sure if there's supposed to be a different, but in any case it makes sense having this scenario as well.